streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -815,19 +815,25 @@ record = null;
         return true;
     }
 
-    private void handleException(final Throwable e) {
-        final StreamsException error = new StreamsException(
+    private void handleException(final Throwable originalException) {
+        handleException(
             String.format(
-                "Exception caught in process. taskId=%s, processor=%s, topic=%s, partition=%d, offset=%d, stacktrace=%s",
+                "Exception caught in process. taskId=%s, processor=%s, topic=%s, partition=%d, offset=%d",
                 id(),
                 processorContext.currentNode().name(),
                 record.topic(),
                 record.partition(),
-                record.offset(),
-                getStacktraceString(e)
+                record.offset()
             ),
-            e
-        );
+            originalException);
+    }
+
+    private void handleException(final String errorMessage, final Throwable originalException) {
```

Comment:
this is an attempt to preserve the error message we create when a handler throws by itself.